olive/passes/olive_pass.py

```diff
@@ -367,6 +367,12 @@ def run(
         """
         Run the pass on the model at a specific point in the search space.
         """
+        if isinstance(model, PyTorchModel) and model.hf_config:
+            hf_config = model.get_hf_model_config().to_dict()
+            input_model_attr = model.model_attributes or {}
+            hf_config.update(input_model_attr)
+            model.model_attributes = hf_config
+
```

Comment:
Is there a reason to no do this in PyTorchModel __init__ ?

Reply:
I agree, I think it might be better to do this in the pytorch model init. 
This way, the serialized input model from the beginning has these model attributes. instead of the `model` object being dynamically updated by the pass run.

Reply:
Oh, we can not put it under pytorch model.

The reason is that: the input model can be hosted in cloud path. But the `get_hf_mode_config` need the model and config with local path which requires to download model to local.  But in AML case, we expect the model and olive_run as all in AML. `Local` just collect the run results.

Reply:
Thanks! that makes sense. Then, I think this way of okay then since all of all information needed is also in the original model. Just not expanded out.